types/three/src/core/Object3D.d.ts

```diff
@@ -0,0 +1,370 @@
+     */
+    receiveShadow: boolean;
+
+    /**
+     * When this is set, it checks every frame if the object is in the frustum of the camera. Otherwise the object gets drawn every frame even if it isn't visible.
+     * @default true
+     */
+    frustumCulled: boolean;
+
+    /**
+     * Overrides the default rendering order of scene graph objects, from lowest to highest renderOrder.
+     * Opaque and transparent objects remain sorted independently though.
+     * When this property is set for an instance of Group, all descendants objects will be sorted and rendered together.
+     * @default 0
+     */
+    renderOrder: number;
+
+    /**
+     * Array with animation clips.
+     * @default []
+     */
+    animations: AnimationClip[];
+
+    /**
+     * An object that can be used to store custom data about the Object3d. It should not hold references to functions as these will not be cloned.
+     * @default {}
+     */
+    userData: { [key: string]: any };
+
+    /**
+     * Custom depth material to be used when rendering to the depth map. Can only be used in context of meshes.
+     * When shadow-casting with a DirectionalLight or SpotLight, if you are (a) modifying vertex positions in
+     * the vertex shader, (b) using a displacement map, (c) using an alpha map with alphaTest, or (d) using a
+     * transparent texture with alphaTest, you must specify a customDepthMaterial for proper shadows.
+     */
+    customDepthMaterial: Material;
+
+    /**
+     * Same as customDepthMaterial, but used with PointLight.
+     */
+    customDistanceMaterial: Material;
+
+    /**
+     * Used to check whether this or derived classes are Object3Ds. Default is true.
+     * You should not change this, as it is used internally for optimisation.
+     */
+    readonly isObject3D: true;
+
+    /**
+     * Calls before rendering object
+     */
+    onBeforeRender: (
+        renderer: WebGLRenderer,
+        scene: Scene,
+        camera: Camera,
+        geometry: BufferGeometry,
+        material: Material,
+        group: Group,
+    ) => void;
+
+    /**
+     * Calls after rendering object
+     */
+    onAfterRender: (
+        renderer: WebGLRenderer,
+        scene: Scene,
+        camera: Camera,
+        geometry: BufferGeometry,
+        material: Material,
+        group: Group,
+    ) => void;
+
+    static DefaultUp: Vector3;
+    static DefaultMatrixAutoUpdate: boolean;
+
+    /**
+     * This updates the position, rotation and scale with the matrix.
+     */
+    applyMatrix4(matrix: Matrix4): void;
+
+    applyQuaternion(quaternion: Quaternion): this;
+
+    setRotationFromAxisAngle(axis: Vector3, angle: number): void;
+
+    setRotationFromEuler(euler: Euler): void;
+
+    setRotationFromMatrix(m: Matrix4): void;
+
+    setRotationFromQuaternion(q: Quaternion): void;
+
+    /**
+     * Rotate an object along an axis in object space. The axis is assumed to be normalized.
+     * @param axis	A normalized vector in object space.
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/three/src/core/Object3D.d.ts;
  Line: 229;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.